addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,249 @@
+        )
+
+        sign = 1 if self.payment_type == 'inbound' else -1
+
+        # The first step is to add withholding lines.
+        for withholding_line in self.l10n_account_withholding_line_ids:
+            # For each withholding line, we need to create a write_off_line with the values of the tax.
+            withholding_tax_values = withholding_line._get_withholding_tax_values()
+
+            if not withholding_line.name:
+                if withholding_line.tax_id.l10n_account_withholding_sequence_id:
+                    withholding_line.name = withholding_line.tax_id.l10n_account_withholding_sequence_id.next_by_id()
+                else:
+                    raise UserError(_('Please enter the withholding number for the tax %(tax_name)s', tax_name=withholding_tax_values['tax_name']))
+
+            tax_account = withholding_tax_values['account']
+            if not tax_account:
+                raise UserError(_('Please define a tax account on the distribution of the tax %(tax_name)s', tax_name=withholding_tax_values['tax_name']))
+
+            payment_vals['write_off_line_vals'].append({
+                'currency_id': self.currency_id.id,
+                'name': f'WH Tax: {withholding_line.name}',
+                'account_id': tax_account,
+                'amount_currency': sign * withholding_line.amount,
+                'balance': sign * self.company_currency_id.round(withholding_line.amount * conversion_rate),
+                'tax_base_amount': withholding_tax_values['tax_base_amount'],
+                'tax_repartition_line_id': withholding_tax_values['tax_repartition_line'],
+                'tax_tag_ids': [Command.set(withholding_tax_values['tag_ids'])],
+            })
+
+        # We also need tax base lines for reporting. We simply create one per unique base amount.
+        balance_sum = amount_currency_sum = 0
+        for base_amount, withholding_lines in self.l10n_account_withholding_line_ids.grouped('base_amount').items():
+            withholding_numbers = ','.join(withholding_lines.mapped('name'))
+            base_amount = sign * base_amount
+            cc_base_amount = self.company_currency_id.round(base_amount * conversion_rate)
+            # The aim is to get one base line with the correct amount, and a counterpart that cancels it.
+            balance_sum += cc_base_amount
+            amount_currency_sum += base_amount
+            payment_vals['write_off_line_vals'].append({
+                'currency_id': self.currency_id.id,
+                'name': f'WH Base: {withholding_numbers}',
+                'tax_ids': [Command.set(withholding_lines.mapped('tax_id').ids)],
+                'account_id': tax_base_account,
+                'balance': cc_base_amount,
+                'amount_currency': base_amount,
+                'tax_tag_ids': [Command.set(withholding_lines._get_withholding_tax_base_tag_ids())],
+            })
+
+        payment_vals['write_off_line_vals'].append({
```

Comment:
Should not we allow to create the withholding entry without the payment itself becoming an accounting entry?  Or are you convinced the accounting entry is absolutely needed?

Reply:
Not sure, it may complicate things. ATM this is the solution that we decided, but I can bring this up with TSB in our meeting tomorrow.